File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        memo = {}
        def rob_house(index):
            if index >= len(nums):
                return 0
            if index in memo:
                return memo[index]
            
            # Option 1: Rob this house and skip the next one
            res = max(nums[index] + rob_house(index + 2), rob_house(index + 1))
            memo[index] = res
            return res

        return rob_house(0)

File: Data Structures & Algorithms/is-subsequence/submission-2.cs
public class Solution
{
	public bool IsSubsequence(string s, string t, int i = 0, int j = 0)
	{
        if (s.Length == 0)
            return true;
            
		if (i >= s.Length || j >= t.Length)
		{
			return false;
		}
		
		if (s[i] == t[j])
		{
            if(i == s.Length - 1)
				return true;
			return IsSubsequence(s, t, i + 1, j + 1);
		}
		else
		{
			return IsSubsequence(s, t, i, j + 1);
		}
		
	}
}